chain/westend-local/default.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package westendlocal
+
+import (
+	"time"
+
+	cfg "github.com/ChainSafe/gossamer/config"
+	"github.com/ChainSafe/gossamer/dot/state/pruner"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/runtime/wasmer"
+)
+
+const (
+	// DefaultBasePathAlice is the default basepath for the westend local alice node
+	DefaultBasePathAlice = "~/.gossamer/westend-local-alice"
+	// DefaultBasePathBob is the default basepath for the westend local bob node
+	DefaultBasePathBob = "~/.gossamer/westend-local-bob"
+	// DefaultBasePathCharlie is the default basepath for the westend local charlie node
+	DefaultBasePathCharlie = "~/.gossamer/westend-local-charlie"
+)
+
+// DefaultAliceConfig returns a westend local node configuration
+func DefaultAliceConfig() *cfg.Config {
+	return &cfg.Config{
+		BaseConfig: cfg.BaseConfig{
+			Name:           "Westend",
+			ID:             "westend_local",
+			BasePath:       DefaultBasePathAlice,
+			ChainSpec:      "./chain/westend-local/westend-local-spec-raw.json",
+			LogLevel:       "info",
+			PrometheusPort: uint32(9876),
+			RetainBlocks:   512,
+			Pruning:        pruner.Archive,
+		},
+		Log: &cfg.LogConfig{
+			Core:    "info",
+			Digest:  "info",
+			Sync:    "info",
+			Network: "info",
+			RPC:     "info",
+			State:   "info",
+			Runtime: "info",
+			Babe:    "info",
+			Grandpa: "info",
+			Wasmer:  "info",
+		},
+		Account: &cfg.AccountConfig{
+			Key:    "alice",
+			Unlock: "",
+		},
+		Core: &cfg.CoreConfig{
+			Role:             common.AuthorityRole,
+			WasmInterpreter:  wasmer.Name,
+			BabeAuthority:    true,
+			GrandpaAuthority: true,
+			GrandpaInterval:  time.Second,
+		},
+		State: &cfg.StateConfig{
+			Rewind: 0,
+		},
+		Network: &cfg.NetworkConfig{
+			Port:              7001,
+			NoBootstrap:       false,
+			NoMDNS:            false,
+			DiscoveryInterval: 10 * time.Second,
+			ProtocolID:        "dot",
+		},
+		RPC: &cfg.RPCConfig{
+			Port: 8545,
+			Host: "localhost",
+			Modules: []string{
+				"system", "author", "chain", "state", "rpc",
+				"grandpa", "offchain", "childstate", "syncstate", "payment"},
+			WSPort: 8546,
+		},
+		Pprof: &cfg.PprofConfig{
+			ListeningAddress: "localhost:6060",
+			BlockProfileRate: 0,
+			MutexProfileRate: 0,
+		},
+		System: &cfg.SystemConfig{
+			SystemName:    "gossamer",
+			SystemVersion: "0.1.0",
+		},
+	}
+}
+
+// DefaultBobConfig returns a westend local node configuration with bob as the authority
+func DefaultBobConfig() *cfg.Config {
```

Comment:
There's a lot of repetition here.  Maybe just return the same config and just modify the base path? I didn't see anything else differ between the configs.

Reply:
yeah, and several ports. I've tidied it up!